File: Python/EstruturaDeRepeticao/exercicio_15.py
"""
A série de Fibonacci é formada pela seqüência 0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55,...
Faça um programa capaz de gerar a série até o n-ésimo termo.
"""

n = int(input("Digite o número de termos da série de Fibonacci desejado: "))
serie = [0, 1]

for i in range(2, n):
    proximo = serie[i - 1] + serie[i - 2]
    serie.append(proximo)

print(f"A série de Fibonacci com {n} termos é:", end=" ")
for numero in serie:
    print(numero, end=" ")

File: Java/EstruturaDeRepeticao/Exercicio15.java
// A série de Fibonacci é formada pela sequência 0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55,...
// Faça um programa capaz de gerar a série até o n−ésimo termo.
package EstruturaDeRepeticao;
import java.util.Scanner;

public class Exercicio15 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Digite o valor de n para gerar a série de Fibonacci até o n-ésimo termo: ");
        int n = scanner.nextInt();

        System.out.println("Série de Fibonacci até o " + n + "º termo:");

        int primeiroTermo = 0;
        int segundoTermo = 1;

        for (int i = 0; i < n; i++) {
            if (i <= 1) {
                System.out.print(i + " ");
            } else {
                int resultado = primeiroTermo + segundoTermo;
                System.out.print(resultado + " ");
                primeiroTermo = segundoTermo;
                segundoTermo = resultado;
            }
        }

        scanner.close();
    }
}
